### PR TITLE
Feat/baggage record UI

### PR DESCRIPTION
**Pull request by oyakovytskyi**
## Describe your changes

- Baggage Recording, Baggage Covering and Locking the Baggage: added UI for new pages
- Added tests for string utils

photo_2024-12-17_20-16-242

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-181)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

photo_2024-12-17_20-16-31
photo_2024-12-23_01-28-01
photo_2024-12-23_01-28-09
photo_2024-12-23_01-28-14

### CODE REVIEW

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/pages/BaggageCover/index.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import useBaggageCover from './useBaggageCover';
+
+import Button from '@/components/Button';
+import Header from '@/components/Header';
+import { iconLabel } from '@/constants/constants';
+import { ButtonGroup, Container } from '@/pages/BaggageRecord/styles';
+import useBaggageRecord from '@/pages/BaggageRecord/useBaggageRecord';
+import StatusIcon from '@/pages/RouteDetails/components/StatusIcon';
+
+const BaggageCoveringPage = (): JSX.Element => {
```
Comment (Parle648) [inline src/pages/BaggageCover/index.tsx]: Why not FC?

File: src/pages/BaggageRecord/index.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import Button from '@/components/Button';
+import Header from '@/components/Header';
+import { COLORS } from '@/constants/colors';
+
+interface PhotoItem {
+  id: number;
+  src: string;
+  description: string;
+}
+// Replace with real BE data
```
Comment (Parle648) [inline src/pages/BaggageRecord/index.tsx]: no comments

**Review by Parle648 — APPROVED** [review]

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/utils/stringUtils.ts [hunk 3]
```diff
@@ -9,3 +9,11 @@ export const getInitials = (name: string = ''): string => {
 export const addPadding = (value: number, zeroesPadding: number): string => {
   return value.toString().padStart(zeroesPadding, '0');
 };
+
+export const truncateString = (str: string, maxLength: number): string => {
```
Comment (lyudad) [inline src/utils/stringUtils.ts:13]: it should be covered by tests

Comment (oyakovytskyi) [reply src/utils/stringUtils.ts:13]: done

File: src/pages/BaggageRecord/useBaggageRecord.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  const [photoList, setPhotoList] = useState<PhotoItem[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [openDeleteModal, setOpenDeleteModal] = useState<boolean>(false);
+  const [selectedPhoto, setSelectedPhoto] = useState<PhotoItem | null>(null);
+  const [isReadyForNextPage, setIsReadyForNextPage] = useState<boolean>(false);
+  const navigate = useNavigate();
+
+  const fetchLuggageData = async (): Promise<void> => {
+    try {
+      const orderId = 4;
```
Comment (lyudad) [inline src/pages/BaggageRecord/useBaggageRecord.ts:48]: с одного места в другой подвинул. мне не нравится хардкорд ордера, это костыль которого я не понимаю

Comment (oyakovytskyi) [reply src/pages/BaggageRecord/useBaggageRecord.ts:48]: This is another PR. Here I can't connect it with the previous pages, we agreed to do it within the next connection pages task. I can leave a comment "Replace with a dynamic order ID when connecting"

**Review by lyudad — APPROVED** [review]

**Pull request merged**